src/users.rs

```diff
@@ -65,6 +65,10 @@ pub struct UserDetails {
     /// A list of external auth identifiers the homeserver has associated with the user.
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub external_ids: Vec<ExternalId>,
+
+    /// Is the account locked
+    #[serde(deserialize_with = "crate::serde::bool_or_uint")]
+    pub locked: bool,
```

Comment:
Can you make this an `Option` so response deserialization doesn't fail for older versions of Synapse that don't return this field?